File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.py
# 1. 끝점 내림차순 정렬
# 2. routes가 빌때까지 순회
# 3.    answer += 1
# 4.    끝점을 뽑고 해당 끝점을 포함하는 모든 선을 pop()
# 5. answer 반환
def solution(routes):
    cnt, answer = 0, 0
    routes.sort(key=lambda x: -x[1])
    
    while routes:
        start, end = routes.pop()
        answer += 1
        while routes and routes[-1][0] <= end <= routes[-1][1]:
            routes.pop()
        
    
    return answer

File: 프로그래머스/lv3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    let answer = 0;
    routes.sort((a, b) => a[1] - b[1])
    while (routes.length) {
        value = routes[0][1]
        routes = routes.filter(r => r[0] > value)
        answer += 1
    }
    return answer;
}